### PR TITLE
Unit Tests for calculator-1.0

### PR DESCRIPTION
**Pull request by y-tst**
Co-Authored-By: Oleksii Yelysieiev <olelysieev@corelogic.com>

### CODE REVIEW

File: src/test/java/calculator/tests/AdditionTest.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package calculator.tests;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class AdditionTest extends GenericTestData {
+
+    @Test(dataProvider = "Values for addition")
+    public void sumOfTwoNumbers(long firstTerm, long secondTerm, long sumResult) {
```
Comment (olelysieev-clgx) [inline src/test/java/calculator/tests/AdditionTest.java]: expectedResult ?

File: src/test/java/calculator/tests/AdditionTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        long result = calculator.sum(firstTerm, secondTerm);
```
Comment (olelysieev-clgx) [inline src/test/java/calculator/tests/AdditionTest.java]: actualResult?

File: src/test/java/calculator/tests/SubtractionTest.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class SubtractionTest extends GenericTestData {
+
+    @Test
+    public void fiveMinusThree() {
+        long result = calculator.sub(5, 3);
+        Assert.assertTrue(result == 2, "Incorrect result of subtraction");
```
Comment (olelysieev-clgx) [inline src/test/java/calculator/tests/SubtractionTest.java]: 'assertEquals' is better in this case

File: src/test/java/calculator/tests/DividingTest.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class DividingTest extends GenericTestData {
+
+    @Test
+    public void fiveMinusThree() {
+        long result = calculator.div(6, 2);
+        Assert.assertTrue(result == 3, "Incorrect result of dividing");
```
Comment (olelysieev-clgx) [inline src/test/java/calculator/tests/DividingTest.java]: assertEquals

File: testng.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (olelysieev-clgx) [inline testng.xml:1]: group and parallelization need to be added alsoin HW5 scope

File: src/test/java/calculator/tests/SubtractionTest.java (hunk 3 above)
Comment (y-tst) [reply src/test/java/calculator/tests/SubtractionTest.java]: Changed. 
Just tried to use different methods.

File: src/test/java/calculator/tests/DividingTest.java (hunk 4 above)
Comment (y-tst) [reply src/test/java/calculator/tests/DividingTest.java]: fixed

File: src/test/java/calculator/listeners/TestListeners.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.Arrays;
+import java.util.logging.Logger;
+
+public class TestListeners implements ITestListener {
+
+    final Logger logger = Logger.getGlobal();
+
+    @Override
+    public void onTestStart(ITestResult result) {
+        logger.info(String.format("Input parameters for %s", result.getName()));
```
Comment (olelysieev-clgx) [inline src/test/java/calculator/listeners/TestListeners.java]: Why not to combine lines 15-16 into one line?

File: src/test/java/calculator/tests/MultiplyingTest.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+import org.testng.annotations.Test;
+
+public class MultiplyingTest extends GenericTestData {
+
+    @Test
+    public void twoMultipleFive() {
+        long actualResult = calculator.mult(2, 5);
+        Assert.assertEquals(actualResult, 10, "Incorrect result of multiplying");
+    }
+
```
Comment (olelysieev-clgx) [inline src/test/java/calculator/tests/MultiplyingTest.java]: Empty line

File: src/test/java/JUnit/JUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import org.junit.runner.RunWith;
+import org.junit.runners.Suite;
+
+@RunWith(Suite.class)
+@Suite.SuiteClasses({
+        SuiteCalculatorSqrtTest.class,
+        SuiteCalculatorPowTest.class
+})
+
+public class JUnitTest extends Calculator {
```
Comment (olelysieev-clgx) [inline src/test/java/JUnit/JUnitTest.java:16]: Why do we need this class with empty methods?

File: src/test/java/calculator/tests/SinusTest.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        double result = round(calculator.sin(Math.PI / 6));
+        Assert.assertEquals(result, 0.5);
+    }
+
+    @Test(groups = "Main Group")
+    public void sin45() {
+        double result = round(calculator.sin(Math.PI / 4));
+        Assert.assertEquals(result, 0.7072);
+    }
+
```
Comment (olelysieev-clgx) [inline src/test/java/calculator/tests/SinusTest.java]: empty line

**Review by olelysieev-clgx — APPROVED** [review]

**Pull request merged**